lisa/tools/cargo.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import PurePath
+from typing import List, Optional, Type, cast
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+from lisa.tools.curl import Curl
+from lisa.tools.echo import Echo
+from lisa.tools.gcc import Gcc
+from lisa.util.process import ExecutableResult
+
+
+class Cargo(Tool):
+    @property
+    def command(self) -> str:
+        return "cargo"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Gcc]
+
+    def _install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        cargo_source_url = "https://sh.rustup.rs"
+        try:
+            # install prerequisites
+            posix_os.install_packages(["build-essential", "cmake"])
+            gcc = self.node.tools[Gcc]
+            gcc_version_info = gcc.get_version().to_dict()
+            self.node.log.info(f"Gcc Version: {gcc_version_info}")
+            # install cargo/rust
```

Comment:
can we put above logic into a separate method like install dependency packages, so that we are clear which steps go wrong.

Reply:
Done